spec/models/user_spec.rb

```diff
@@ -1087,4 +1087,25 @@
     end
   end
 
+  describe ".info_requests_count" do
```

Comment:
This passes against develop.

```sh
git fetch origin
git checkout develop
git checkout origin/patch-rails4-counter-cache-bug -- spec/models/user_spec.rb
bundle exec rspec spec/models/user_spec.rb
# …
128 examples, 0 failures
```

I think a spec that more closely follows [the failing example](https://gist.github.com/jimjh/6057340#file-bug_10865-rb-L77) would be better.